pkg/kubevirt/client.go

```diff
@@ -180,3 +213,126 @@ func (c *client) DeleteDataVolume(ctx context.Context, namespace string, name st
 func (c *client) GetDataVolume(ctx context.Context, namespace string, name string) (*cdiv1.DataVolume, error) {
 	return c.cdiClient.CdiV1beta1().DataVolumes(namespace).Get(ctx, name, metav1.GetOptions{})
 }
+
+func (c *client) CreateVolumeSnapshot(ctx context.Context, namespace, name, claimName, snapshotClassName string) (*snapshotv1.VolumeSnapshot, error) {
+	snapshotClassName, err := c.getSnapshotClassNameFromVolumeName(ctx, namespace, claimName, snapshotClassName)
```

Comment:
when the infra snapshotClassName is provided as a parameter ```snapshotClassName := req.Parameters[infraSnapshotClassNameParameter]```, then it doesn't make sense to me why we're trying to match an infra snapshotClass by looking up all the snapshotClasses and attempting to make a best effort match.

Essentially, if we're told the snapshotClass to use, why would we not just be using that instead of trying to detect the snapshotClass from the volume somehow.

Reply:
Good point, I just need to detect that if we have a provided volume snapshot class, it can match the volume we are trying to snapshot. The thing I am matching against is that the provider associated with the volume we are snapshotting actually matches the volume snapshot we are attempting to use. If we are provided one and it doesn't match the ones associated with the provider of the volume, we should error before attempting a snapshot. The code does not do that and I will fix it.